spec/features/question_form_spec.rb

```diff
@@ -51,4 +51,25 @@
       end
     end
   end
+
+  scenario "audio upload works", js: true do
+    visit new_question_path(locale: "en", mode: "m", mission_name: mission.compact_name)
+    fill_in "Code", with: "AQuestion"
+    fill_in "Title", with: "Jay's"
+    select "Text", from: "Type"
+
+    attach_file("Audio Prompt", fixture("powerup.mp3"))
+    click_on "Save"
+
+    visit question_path(locale: "en", mode: "m",
+      mission_name: mission.compact_name, id: Question.last.id)
+
+    expect(page).to have_content("powerup.mp3")
+  end
+
+  private
+
+  def fixture(name)
+    File.expand_path("../../fixtures/media/audio/#{name}", __FILE__)
+  end
```

Comment:
Another question -- where did this code come from? Did you write it or copy it from elsewhere? If the latter, nice find! Does copying code raise any flags for you? Are there other approaches you might have taken?

Reply:
After going through the codebase, I discovered that this was the code style for the app on expanding fixture files plus it required less code

There are different ways I would have gone about this: using Rails.root and appending it to the path or using join to build the path

Reply:
Gotcha. What I'd like you to consider is if you can refactor this into a helper file (maybe general_spec_helpers.rb) so that it doesn't need to be copied into each file that needs to load a fixture.

If you do so, you'll need to change it to the `Rails.root.join` version so that it works no matter what subdirectory the current spec file is in.

Reply:
Wouldn't it be better if we just refactored the fixture methods themselves in the tests instead of moving it to a helper. I'm tilting towards this direction because the different files are expanding different directories and I'm thinking how it'd be passing in various arguments

Reply:
You can write a general helper method that takes a sub-path under the fixtures directory and returns the file. There is already a media_fixture method just for media fixtures. That method could call your new method.